ethereum/core/src/test/java/org/hyperledger/besu/ethereum/privacy/PrivateStateRootResolverTest.java

```diff
@@ -0,0 +1,180 @@
+import org.hyperledger.besu.ethereum.core.TransactionReceipt;
+import org.hyperledger.besu.ethereum.privacy.storage.PrivacyGroupHeadBlockMap;
+import org.hyperledger.besu.ethereum.privacy.storage.PrivateBlockMetadata;
+import org.hyperledger.besu.ethereum.privacy.storage.PrivateStateStorage;
+import org.hyperledger.besu.ethereum.privacy.storage.PrivateTransactionMetadata;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class PrivateStateRootResolverTest {
+
+  private static final BlockDataGenerator BLOCK_GENERATOR = new BlockDataGenerator();
+  private static MutableBlockchain BLOCKCHAIN;
+
+  private static final Hash pmt1StateHash =
+      Hash.fromHexString("0x37659019840d6e04e740614d1ad93d62f0d9d7cc423b2178189f391db602a6a6");
+  private static final Hash pmt2StateHash =
+      Hash.fromHexString("0x12d390c87b405e91523b5829002bf90095005366eb9aa168ff8a18540902e410");
+  private static final Bytes32 privacyGroupId =
+      Bytes32.wrap(Bytes.fromBase64String("A1aVtMxLCUHmBVHXoZzzBgPbW/wj5axDpW9X8l91SGo="));
+  private static final Bytes32 failingPrivacyGroupId =
+      Bytes32.wrap(Bytes.fromBase64String("Ko2bVqD+nNlNYL5EE7y3IdOnviftjiizpjRt+HTuFBs="));
+
+  private PrivateStateStorage privateStateStorage;
+
+  @BeforeClass
+  public static void setupClass() {
+    BLOCKCHAIN = InMemoryStorageProvider.createInMemoryBlockchain(BLOCK_GENERATOR.genesisBlock());
+    for (int i = 1; i <= 69; i++) {
+      final BlockDataGenerator.BlockOptions options =
+          new BlockDataGenerator.BlockOptions()
+              .setBlockNumber(i)
+              .setParentHash(BLOCKCHAIN.getBlockHashByNumber(i - 1).get());
+      final Block block = BLOCK_GENERATOR.block(options);
+      final List<TransactionReceipt> receipts = BLOCK_GENERATOR.receipts(block);
+      BLOCKCHAIN.appendBlock(block, receipts);
+    }
+  }
+
+  @Before
+  public void setUp() {
+    privateStateStorage = InMemoryStorageProvider.createInMemoryPrivateStateStorage();
+  }
+
+  @Test
+  public void mustResolveStateRootIfChainHeadIsNotCommitted() {
+    final BlockDataGenerator.BlockOptions options =
+        new BlockDataGenerator.BlockOptions()
+            .setBlockNumber(BLOCKCHAIN.getChainHeadBlockNumber())
+            .setParentHash(BLOCKCHAIN.getChainHeadHash());
+    final Block block = BLOCK_GENERATOR.block(options);
+    final PrivateStateRootResolver privateStateRootResolver =
+        new PrivateStateRootResolver(privateStateStorage);
+    assertThat(
+            privateStateRootResolver.resolveLastStateRoot(
+                privacyGroupId, block.getHeader().getHash()))
+        .isEqualTo(PrivateStateRootResolver.EMPTY_ROOT_HASH);
+  }
+
+  @Test
+  public void ifNoCommitmentForPrivacyGroupExistsReturnEmptyRootHash() {
```

Comment:
NIT: `resolveEmptyRootHashWhenNoCommitmentForPrivacyGroupExists` would match the most common naming pattern used for unit tests in Besu.